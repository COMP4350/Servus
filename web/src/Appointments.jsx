import React from "react";

const Appointment = () => {
  return <div>This is the Appointment page</div>;
};

export default Appointment;
