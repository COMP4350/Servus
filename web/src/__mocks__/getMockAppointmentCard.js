function getMockAppointmentCard() {
    return {
        buyer: 'Risto',
        provider: 'Arveend',
        service_id: '1001',
        created_at: new Date(100),
        booked_time: new Date(500),
    }
}

export const mockAppointmentCard = getMockAppointmentCard();