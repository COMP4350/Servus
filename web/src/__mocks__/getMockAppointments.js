function getMockAppointments() {
    return [
        {
            buyer: 'Risto',
            provider: 'Arveend',
            service_id: '1001',
            created_at: new Date(100),
            booked_time: new Date(500),
        },
        {
            buyer: 'Arveend',
            provider: 'Risto',
            service_id: '1002',
            created_at: new Date(700),
            booked_time: new Date(1200),
        },
        {
            buyer: 'Caden',
            provider: 'Andy',
            service_id: '1003',
            created_at: new Date(1400),
            booked_time: new Date(1900),
        },
    ]
}

function getMockAppointmentsCookie() {
    return {
        username: 'testusername',
    }
}

export const mockAppointments = getMockAppointments();
export const mockAppointmentsCookie = getMockAppointmentsCookie();