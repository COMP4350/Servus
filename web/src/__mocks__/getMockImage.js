function getMockImage() {
    return {
        ownerUsername: 'riki',
        uploadDate: new Date(2012, 1, 1),
        imageName: 'example_image.jpeg',
        imageData: 'the/link/to/data',
        profilePicture: false,
    };
}

export const mockImage = getMockImage();
