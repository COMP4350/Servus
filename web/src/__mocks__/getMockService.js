function getMockService() {
    return {
        _id: '1001',
        name: "Risto's Painting",
        provider: 'Risto',
        description: 'Risto paints you',
        cost: 100,
        duration: '0100',
        availability: [{ weekday: 1, start_time: '0800', end_time: '1800' }],
        ratings: [{ username: 'arvind', rating: 5 }],
    };
}

export const mockService = getMockService();
