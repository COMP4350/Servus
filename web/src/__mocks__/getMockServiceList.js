function getMockServiceList() {
    return [
        {
            name: "Risto's Painting",
            provider: 'Risto',
            description: 'Risto paints you',
            cost: 100,
            duration: 60,
        },
        {
            name: "Reesto's Painting",
            provider: 'Reesto',
            description: 'Reesto paints yee',
            cost: 120,
            duration: 80,
        },
        {
            name: "Rasta's Painting",
            provider: 'Rasta',
            description: 'Rasta paints yaa',
            cost: 150,
            duration: 90,
        },
    ];
}

export const mockServiceList = getMockServiceList();
