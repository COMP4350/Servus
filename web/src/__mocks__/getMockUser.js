function getMockUser() {
    return {
        username: 'testuser',
        firstName: 'test',
        lastName: 'user',
        password: '123',
        bio: 'im a test user',
    };
}

export const mockUser = getMockUser();
