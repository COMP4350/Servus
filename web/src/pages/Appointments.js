import React, { useEffect, useState } from 'react';
import axios from 'axios';
import Calendar from '../components/Calendar';
import { useCookies } from 'react-cookie';
import { useHistory } from 'react-router-dom';

const Appointment = () => {
    const [appointments, setAppointments] = useState();
    const [cookies] = useCookies();
    const history = useHistory();
    const fetchAppointments = async username => {
        //this gets appointments where the user is both the buyer and provider
        const res = await axios.get(`/appointment/${username}`);
        setAppointments(res.data.result);
        console.log(appointments);
    };
    //fetch appointments
    useEffect(() => {
        if (cookies.username) fetchAppointments(cookies.username);
        else history.push('/login');
    }, []);
    return (
        <div>
            {appointments ? (
                <Calendar appointments={appointments}></Calendar>
            ) : null}
        </div>
    );
};

export default Appointment;
