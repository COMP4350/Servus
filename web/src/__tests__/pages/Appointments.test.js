import React from 'react';
import { act, render } from '@testing-library/react';
import Appointments from '../../pages/Appointments';
import { BrowserRouter } from 'react-router-dom';
import { mockAppointments } from '../../__mocks__/getMockAppointmentCard';
import mockAxios from 'axios';

const blankComponent = () => {
    return <div></div>;
};

jest.mock('react-cookie', () => {
    return {
        __esModule: true,
        useCookies: () => {
            return [{ username: 'yoyo' }];
        },
    };
});

jest.mock('../../components/Calendar', () => {
    return {
        __esModule: true,
        default: () => {
            return blankComponent();
        },
    };
});

describe('Appointments', () => {
    it('renders correctly', async () => {
        const promise = Promise.resolve();

        mockAxios.get.mockImplementationOnce(() =>
            Promise.resolve({
                data: { result: mockAppointments },
            })
        );

        const { asFragment } = render(
            <BrowserRouter>
                <Appointments />
            </BrowserRouter>
        );

        expect(asFragment()).toMatchSnapshot();
        expect(mockAxios.get).toHaveBeenCalledWith('/appointment/yoyo');
        expect(mockAxios.get).toHaveBeenCalledTimes(1);

        await act(() => promise);
    });
});
