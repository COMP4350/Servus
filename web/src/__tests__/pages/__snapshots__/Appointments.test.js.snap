// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appointments renders correctly 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-container-1"
  />
</DocumentFragment>
`;
